From 1c007b659086bb00b5713cb942d67555062d70c5 Mon Sep 17 00:00:00 2001
From: Ricky Zhang <ricky.zhang@mediatek.com>
Date: Mon, 4 Dec 2017 14:34:55 +0800
Subject: [PATCH] video-fomat: fix MT21 info

MT21 fourcc is MM21.
Test: test ok

Signed-off-by: Ricky Zhang <ricky.zhang@mediatek.com>
CR-Id: AUTO00012426
---
 gst-libs/gst/video/video-format.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/video/video-format.c b/gst-libs/gst/video/video-format.c
index 3220106..e5bf55c 100644
--- a/gst-libs/gst/video/video-format.c
+++ b/gst-libs/gst/video/video-format.c
@@ -3574,7 +3574,7 @@ static const VideoFormat formats[] = {
       PSTR2222, PLANE0123, OFFS0, SUB4444, PACK_A444_10LE),
   MAKE_YUV_FORMAT (NV61, "raw video", GST_MAKE_FOURCC ('N', 'V', '6', '1'),
       DPTH888, PSTR122, PLANE011, OFFS010, SUB422, PACK_NV61),
-  MAKE_YUV_FORMAT (MT21, "raw video", GST_MAKE_FOURCC ('M', 'T', '2', '1'),
+  MAKE_YUV_FORMAT (MT21, "raw video", GST_MAKE_FOURCC ('M', 'M', '2', '1'),
       DPTH888, PSTR122, PLANE011, OFFS001, SUB420, PACK_NV12),
 };
 
@@ -3804,6 +3804,8 @@ gst_video_format_from_fourcc (guint32 fourcc)
       return GST_VIDEO_FORMAT_IYU1;
     case GST_MAKE_FOURCC ('A', 'Y', '6', '4'):
       return GST_VIDEO_FORMAT_AYUV64;
+    case GST_MAKE_FOURCC ('M', 'M', '2', '1'):
+      return GST_VIDEO_FORMAT_MT21;
     default:
       return GST_VIDEO_FORMAT_UNKNOWN;
   }
@@ -3825,6 +3827,9 @@ gst_video_format_from_string (const gchar * format)
 
   g_return_val_if_fail (format != NULL, GST_VIDEO_FORMAT_UNKNOWN);
 
+  if (strcmp ("MT21", format) == 0)
+    return GST_VIDEO_FORMAT_MT21;
+
   for (i = 0; i < G_N_ELEMENTS (formats); i++) {
     if (strcmp (GST_VIDEO_FORMAT_INFO_NAME (&formats[i].info), format) == 0)
       return GST_VIDEO_FORMAT_INFO_FORMAT (&formats[i].info);
-- 
1.9.1

